cpp/src/knn/knn_regress_mg.cu

```diff
@@ -57,17 +57,30 @@ using namespace knn_common;
 
 void knn_regress(raft::handle_t &handle,
                  std::vector<Matrix::Data<float> *> *out,
-                 std::vector<Matrix::Data<int64_t> *> *out_I,
-                 std::vector<Matrix::floatData_t *> *out_D,
                  std::vector<Matrix::floatData_t *> &idx_data,
                  Matrix::PartDescriptor &idx_desc,
                  std::vector<Matrix::floatData_t *> &query_data,
                  Matrix::PartDescriptor &query_desc,
                  std::vector<std::vector<float *>> &y, bool rowMajorIndex,
                  bool rowMajorQuery, int k, int n_outputs, size_t batch_size,
                  bool verbose) {
-  opg_knn(handle, out, out_I, out_D, idx_data, idx_desc, query_data, query_desc,
-          y, rowMajorIndex, rowMajorQuery, k, n_outputs, batch_size, verbose);
+  opg_knn_param params;
```

Comment:
Just for future-proofing some of our code, we might want to consider adding a constructor to this object since it is being instantiated in so many different places now.